clang/test/SemaSYCL/redeclaration-attribute-propagation.cpp

```diff
@@ -0,0 +1,72 @@
+// RUN: %clang_cc1 %s -fsyntax-only -fsycl -fsycl-is-device -triple spir64 -verify
+// RUN: %clang_cc1 %s -fsyntax-only -fsycl -fsycl-is-device -triple spir64 -DTRIGGER_ERROR -verify
+// RUN: %clang_cc1 %s -fsyntax-only -ast-dump -fsycl -fsycl-is-device -triple spir64 | FileCheck %s
+
+#ifndef TRIGGER_ERROR
+//first case - good case
+[[intelfpga::no_global_work_offset]] // expected-no-diagnostics
+void
+func1();
+
+[[intelfpga::max_work_group_size(4, 4, 4)]] void func1();
+
+[[cl::reqd_work_group_size(2, 2, 2)]] void func1() {}
+
+#else
+//second case - expect error
+[[intelfpga::max_work_group_size(4, 4, 4)]] void
+func2();
+
+[[cl::reqd_work_group_size(8, 8, 8)]] // expected-note {{conflicting attribute is here}}
+void
+func2() {}
```

Comment:
Just to confirm: this error is not detected by the changes in this patch, right?